pkg/providers/vault/client.go

```diff
@@ -48,11 +71,148 @@ func NewSecretClient(config SecretConfig) (pkg.SecretClient, error) {
 		config.retryWaitPeriodTime = retryTimeDuration
 	}
 
-	return Client{
+	secretClient := Client{
 		HttpConfig: config,
 		HttpCaller: httpClient,
-	}, nil
+		lc:         lc,
+	}
+
+	if bkgCtx == nil {
+		bkgCtx = context.Background()
+	}
```

Comment:
Context should be required IMO.  Don't create a default if it isn't passed in.  Set a guard to ensure the reference provided isn't nil.

Reply:
ok